pydoctor/epydoc/markup/restructuredtext.py

```diff
@@ -155,28 +157,32 @@ class _EpydocReader(StandaloneReader):
     and appends them to a list.
     """
 
-    def __init__(self, errors):
+    def __init__(self, errors: List[ParseError]):
         self._errors = errors
         StandaloneReader.__init__(self)
 
-    def get_transforms(self):
+    def get_transforms(self) -> List[Transform]:
         # Remove the DocInfo transform, to ensure that :author: fields
         # are correctly handled.
         return [t for t in StandaloneReader.get_transforms(self)
                 if t != docutils.transforms.frontmatter.DocInfo]
 
-    def new_document(self):
+    def new_document(self) -> docutils.nodes.document:
         document = new_document(self.source.source_path, self.settings)
         # Capture all warning messages.
         document.reporter.attach_observer(self.report)
         # Return the new document.
         return document
 
-    def report(self, error):
-        try: is_fatal = int(error['level']) > 2
-        except: is_fatal = True
-        try: linenum = int(error['line'])
-        except: linenum = None
+    def report(self, error: docutils.nodes.system_message) -> None:
+        try:
+            is_fatal = int(error['level']) > 2
+        except:
```

Comment:
Maybe 
```suggestion
        except Exception:
```

Reply:
Indeed we shouldn't be catching everything.

After reading a bit of the `docutils` code, I don't think `level` could be undefined or something other than an integer, so I'll just remove this overly cautious piece of code.